### PR TITLE
Error handler for auth pages

### PR DESCRIPTION
**Pull request by kas1qqqq**
**400ths errors**
SCR-20230618-uils
**500ths errors**
SCR-20230621-lkgs

### CODE REVIEW

**Comment by kas1qqqq** [comment]
[Wrong error handling for sign in/sign up pages](https://trello.com/c/f9Su31G3/117-wrong-error-handling-for-sign-in-sign-up-pages)

**Comment by kas1qqqq** [comment]
[Wrong error handling for sign in/sign up pages](https://trello.com/c/f9Su31G3/117-wrong-error-handling-for-sign-in-sign-up-pages)

**Comment by kas1qqqq** [comment]
[Wrong error handling for sign in/sign up pages](https://trello.com/c/f9Su31G3/117-wrong-error-handling-for-sign-in-sign-up-pages)

File: src/components/elements/Auth/Auth.tsx [hunk 1]
```diff
@@ -1,23 +1,35 @@
...
+}
+
 interface AuthProps {
     keyword?: string;
     isSubmitting: boolean;
     children: ReactNode;
     title: string;
     description: string;
-    error?: string;
+    error?: ErrorProps | any;
```
Comment (karinabarinova) [inline src/components/elements/Auth/Auth.tsx:28]: Is there any specific reason why we have `any` here?

File: src/components/elements/Error/Error.tsx [hunk 2]
```diff
@@ -1,18 +1,24 @@
-import React from "react";
+import { FC } from "react";
 
 import { Text } from "@elements/Text";
 import { useCssClasses } from "@utils/getClassnames";
 
 import styles from "./Error.module.scss";
 
 type ErrorProps = {
+    semiBold?: true | false;
```
Comment (karinabarinova) [inline src/components/elements/Error/Error.tsx:9]:
```suggestion
    semiBold?: boolean;
```

File: src/components/elements/Title/Title.tsx [hunk 3]
```diff
@@ -8,12 +8,24 @@ interface TitleProps {
     level?: 1 | 2 | 3 | 4 | 5;
     children: ReactNode;
     color?: "darkTextColor" | "lightTextColor" | "mostLightColor";
+    semiBold?: true | false;
```
Comment (karinabarinova) [inline src/components/elements/Title/Title.tsx:11]:
```suggestion
    semiBold?: boolean;
```

File: src/components/modules/ResetPasswordSection/ResetPasswordForm.tsx [hunk 4]
```diff
@@ -54,15 +54,15 @@ export const ResetPasswordForm: FC = () => {
             description="Enter your email address and we will send you instructions to reset your password."
             onSubmit={formik.handleSubmit}
         >
-            {textFields.map(({ type, placeholder, value, error, name }) => (
+            {textFields.map(({ type, placeholder, value, error: err, name }) => (
```
Comment (karinabarinova) [inline src/components/modules/ResetPasswordSection/ResetPasswordForm.tsx:57]: Why are we renaming the error field here?

File: src/components/modules/SignInSection/SignInForm.tsx [hunk 5]
```diff
@@ -22,7 +23,11 @@ export const SignInForm: FC = () => {
     const dispatch = useAppDispatch();
     const router = useRouter();
 
-    const [loginRequest, { data, isError }] = useLoginMutation();
+    const [loginRequest, { data, error, isLoading }] = useLoginMutation();
+
+    const isFetchBaseQueryError = (err: unknown): err is FetchBaseQueryError => {
```
Comment (karinabarinova) [inline src/components/modules/SignInSection/SignInForm.tsx:28]: Let's move this function to a util file.

File: src/components/modules/SignUpSection/SignUpForm.tsx [hunk 6]
```diff
@@ -18,7 +19,11 @@ interface FormValues {
 }
 
 export const SignUpForm: FC = () => {
-    const [registerRequest] = useRegisterMutation();
+    const [registerRequest, { error, isLoading }] = useRegisterMutation();
+
+    const isFetchBaseQueryError = (err: unknown): err is FetchBaseQueryError => {
```
Comment (karinabarinova) [inline src/components/modules/SignUpSection/SignUpForm.tsx:24]: Once the function has been moved to a util file, we can reuse it in this file as well

**Review by karinabarinova — CHANGES_REQUESTED** [review]

File: src/components/modules/ResetPasswordSection/ResetPasswordForm.tsx (hunk 4 above)
Comment (kas1qqqq) [reply src/components/modules/ResetPasswordSection/ResetPasswordForm.tsx:57]: Case in sign in sign up components already exist error object from rtk query

File: src/components/elements/Auth/Auth.tsx (hunk 1 above)
Comment (kas1qqqq) [reply src/components/elements/Auth/Auth.tsx:28]: Ts can’t define what is will be come from server, maybe you can help fix it

File: src/components/modules/SignInSection/SignInForm.tsx (hunk 5 above)
Comment (kas1qqqq) [reply src/components/modules/SignInSection/SignInForm.tsx:28]: Ok

File: src/components/modules/SignUpSection/SignUpForm.tsx (hunk 6 above)
Comment (kas1qqqq) [reply src/components/modules/SignUpSection/SignUpForm.tsx:24]: Ok

File: src/components/elements/Auth/Auth.tsx (hunk 1 above)
Comment (kas1qqqq) [reply src/components/elements/Auth/Auth.tsx]: In the last commit, I reworked this logic

**Review by karinabarinova — APPROVED** [review]

**Review by DevMoroh — APPROVED** [review]

**Pull request merged**